src/presenced.cpp

```diff
@@ -281,10 +291,18 @@ Client::reconnect(const std::string& url)
                 return promise::Error("No valid URL provided and current URL is not valid");
         }
 
-        setConnState(kConnecting);
-        return retry("presenced", [this](int no)
+        auto wptr = weakHandle();
+        if (mRetryCtrl)
         {
+            mRetryCtrl->abort();
+        }
+        mRetryCtrl.reset(createRetryController("presenced", [wptr, this](int no) -> Promise<void>
+        {
+            if (wptr.deleted())
+                return promise::Error("destroyed");
```

Comment:
That's what we do - `mRetryCtrl->abort()`, or did I misunderstand what you mean?